core/src/core_tests/child_workflows.rs

```diff
@@ -97,3 +103,68 @@ async fn cancel_child_workflow() {
     wfm.process_all_activations().await.unwrap();
     wfm.shutdown().await.unwrap();
 }
+
+#[rstest::rstest]
+#[case::abandon(ChildWorkflowCancellationType::Abandon)]
+#[case::try_cancel(ChildWorkflowCancellationType::TryCancel)]
+#[case::wait_cancel_completed(ChildWorkflowCancellationType::WaitCancellationCompleted)]
+#[tokio::test]
+async fn cancel_child_workflow_lang_thinks_not_started_but_is(
```

Comment:
Test cancellation of not started child too? Is this interesting enough to test?

Reply:
Test cancellation of already completed child, core should ignore this (we do the same for activities AFAIR).

Reply:
Added